### PR TITLE
#4387 Enhance Minimum sum partition problem implementation

### PR DESCRIPTION
**Pull request by Manan-09**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

Resolves [4387](https://github.com/TheAlgorithms/Java/issues/4387)

Testcase cover 100% code

Screenshot 2023-09-22 at 2 50 15 AM

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/MinimumSumPartition.java [hunk 1]
```diff
@@ -17,73 +19,31 @@
  */
 public class MinimumSumPartition {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/MinimumSumPartition.java:20]:
```suggestion
public final class MinimumSumPartition {
```

It is a _utility class_, so there are no reasons to _extend_ it.

File: src/test/java/com/thealgorithms/dynamicprogramming/MinimumSumPartitionTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.thealgorithms.dynamicprogramming;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class MinimumSumPartitionTest {
+
```
Comment (vil02) [inline src/test/java/com/thealgorithms/dynamicprogramming/MinimumSumPartitionTest.java:8]:
```suggestion
```

File: src/main/java/com/thealgorithms/dynamicprogramming/MinimumSumPartition.java [hunk 3]
```diff
@@ -17,73 +19,31 @@
...
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/MinimumSumPartition.java:21]:
```suggestion
```

**Review by vil02 — CHANGES_REQUESTED** [review]
I would suggest to add the examples from the description to the test file.

Could you also update the description of the problem? At the moment is not very instructive.

Does it work for input array having negative numbers? This would need some explanation.

**Comment by Manan-09** [comment]
@vil02  This implementation is for non-negative numbers only.

File: src/main/java/com/thealgorithms/dynamicprogramming/MinimumSumPartition.java [hunk 4]
```diff
@@ -1,89 +1,50 @@
 package com.thealgorithms.dynamicprogramming;
 
-// Partition a set into two subsets such that the difference of subset sums is minimum
+import java.util.Arrays;
 
 /*
-Input:  arr[] = {1, 6, 11, 5}
-Output: 1
+Given an array of non-negative integers , partition the array in two subset that
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/MinimumSumPartition.java:6]: If it is only for non-negative numbers, please check the input array and throw `IllegalArgumentException` when necessary, cf. https://github.com/TheAlgorithms/Java/pull/4368#issuecomment-1719945280

File: src/test/java/com/thealgorithms/dynamicprogramming/MinimumSumPartitionTest.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package com.thealgorithms.dynamicprogramming;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (vil02) [inline src/test/java/com/thealgorithms/dynamicprogramming/MinimumSumPartitionTest.java:3]:
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/dynamicprogramming/MinimumSumPartitionTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        int[] array = {};
+        assertEquals(0, MinimumSumPartition.minimumSumPartition(array));
+    }
+
+    @Test
+    public void testMinimumSumPartitionThrowsForNegativeArray() {
+        int[] array = {4, 1, -6, 7};
+        assertThrows(IllegalArgumentException.class, () -> {
+            MinimumSumPartition.minimumSumPartition(array);
+        });
```
Comment (vil02) [inline src/test/java/com/thealgorithms/dynamicprogramming/MinimumSumPartitionTest.java:44]:
```suggestion
        assertThrows(IllegalArgumentException.class, () -> { MinimumSumPartition.minimumSumPartition(array); });
```

**Review by vil02 — CHANGES_REQUESTED** [review]

**Review by vil02 — APPROVED** [review]

**Pull request merged**